Sources/App/CRP/CRPProcess.swift

```diff
@@ -0,0 +1,122 @@
+        case missingParameter(String)
+    }
+
+    static func apiRequest(request: Request, github: GitHubService, jira: JiraService) throws -> Future<Response> {
+        let repo: String = try Option.repo.get(from: request)
+        let branch: String = try Option.branch.get(from: request)
+        let skipTicket: Bool = (try? Option.skipTicket.get(from: request)) ?? false
+        let skipFixVersion: Bool = (try? Option.skipFixVersion.get(from: request)) ?? false
+
+        guard let repoMapping = RepoMapping.all[repo.lowercased()] else {
+            throw CRPProcess.Error.invalidParameter(
+                key: Option.repo.rawValue,
+                value: repo,
+                expected: RepoMapping.all.keys.joined(separator: "|")
+            )
+        }
+
+        let release = try GitHubService.Release(
+            repo: repoMapping.repository,
+            branch: branch
+        )
+
+        return try github.changelog(for: release, on: request)
+            .catchError(.capture())
+            .flatMap { (commitMessages: [String]) -> Future<(JiraService.CreatedIssue?, JiraService.FixedVersionReport?)> in
+
+                let jiraVersionName = repoMapping.crp.jiraVersionName(release)
+                let changelogSections = ChangelogSection.makeSections(from: commitMessages, for: release)
+
+                // Create CRP ticket, unless skipped
+                let createTicket: Future<JiraService.CreatedIssue?>
+                if skipTicket {
+                    createTicket = request.future(nil)
+                } else {
+                    // Create CRP Issue
+                    let crpIssue = JiraService.makeCRPIssue(
+                        jiraBaseURL: jira.baseURL,
+                        crpProjectID: JiraService.crpProjectID,
+                        crpConfig: repoMapping.crp,
+                        release: release,
+                        changelog: jira.document(from: changelogSections)
+                    )
+
+                    createTicket = try jira.create(issue: crpIssue, on: request)
+                        .map { Optional.some($0) }
+                        .catchError(.capture())
+                }
+
+                return createTicket
+                    .flatMap { (crpIssue: JiraService.CreatedIssue?) -> Future<(JiraService.CreatedIssue?, JiraService.FixedVersionReport?)> in
+                        guard !skipFixVersion else {
+                            return request.future( (crpIssue, nil) )
+                        }
+                        // Create JIRA versions on each board then set Fixed Versions to that new version on each board's ticket included in Changelog
+                        return try jira.createAndSetFixedVersions(
+                            changelogSections: changelogSections,
+                            versionName: jiraVersionName,
+                            on: request
+                        ).map { (crpIssue, Optional.some($0)) }
```

Comment:
ditto maybe 🤔

Reply:
`.map((crpIssue, Optional.some))`

Reply:
Mmm not sure if that works inside a tuple? Will try 👍

Reply:
Tried, it doesn't (which makes sense, signatures don't match). Could use `map(Optional.some).map { (crpIssue, $0) }` instead, but probably not worth it

Reply:
ah yeah, thanks for trying! not a big deal 👍